src/App.js

```diff
@@ -1,12 +1,101 @@
-import React from 'react';
+//Functions
+import React, {useEffect, useState} from 'react';
 import './App.css';
 
+//Components
+import PokemonCard from "./components/PokemonCard/PokemonCard";
+import Button from "./components/Button/Button";
+
+//Assets
+import logo from './assets/logo.png'
+import axios from "axios";
+
+//Omdat ik vast nog weleens in dit document ga terugkijken, ga ik er veel tekst bijzetten!
+//Om te veel dubbele stukken tekst te voorkomen, heb ik als eerstre informatie in pokemoncard.js gezet
+//Hier zal ik de asynchrone functie dan ook niet meer bespreken
+
 function App() {
-  return (
-    <div>
-      Begin hier met de opdracht!
-    </div>
-  );
+    //Als eerste wordt de state geinitialiseerd als een link met de 'basis' api
+    //'endpoint' maakt het mogelijk om dit stukje informatie overal te gebruiken in dit document
+    //'setEndpoint' maakt het mogelijk om de link te veranderen (bijv. met de buttons)
+    const [endpoint, setEndpoint] = useState('https://pokeapi.co/api/v2/pokemon');
+    //de state wordt als eerste geinitialiseerd als een leeg object.
+    //'pokemons' maakt het mogelijk om die data ook buiten de scope van de functie te gebruiken (dus in de return)
+    //'setpokemons' zorgt ervoor dat de waarde van pokemon verandert kan worden
+    const [pokemons, setPokemons] = useState([]);
```

Comment:
Dit zal nooit een array zijn, gebruik daarom liever `undefined` als default waarde. Dan kun je ook een wat betere, en makkelijkere, check doen of de waarde al geset is of niet

```suggestion
    const [pokemons, setPokemons] = useState();
```